File: Leetcode_Medium/_853. Car Fleet/Solution.py
class Solution:
    def carFleet(self, target , position, speed):
        # Create a list of tuples to store position and time-to-reach pairs
        cars = sorted(zip(position, speed), reverse=True)
        
        # Initialize fleet count and previous arrival time
        fleets = 0
        prevTime = 0
        
        # Iterate through the cars
        for pos, spd in cars:
            # Calculate time to reach the target
            time = (target - pos) / spd
            
            # If the current car takes more time to reach, it forms a new fleet
            if time > prevTime:
                fleets += 1
                prevTime = time
        
        return fleets

File: Leetcode_Medium/_853. Car Fleet/Solution.java
import java.util.*;

class Solution {
    public int carFleet(int target, int[] position, int[] speed) {
        // Create a TreeMap to store position and time-to-reach pairs
        TreeMap<Integer, Double> map = new TreeMap<>();
        
        // Populate the TreeMap with position and time-to-reach pairs
        for (int i = 0; i < position.length; i++) {
            map.put(-position[i], (double)(target - position[i]) / speed[i]);
        }
        
        // Initialize fleet count and previous arrival time
        int fleets = 0;
        double prevTime = 0;
        
        // Iterate through the TreeMap
        for (double time : map.values()) {
            // If the current car takes more time to reach, it forms a new fleet
            if (time > prevTime) {
                fleets++;
                prevTime = time;
            }
        }
        
        return fleets;
    }
}
